ext/pdo/pdo_stmt.c

```diff
@@ -1716,13 +1736,22 @@ PHP_METHOD(PDOStatement, getColumnMeta)
 
 /* {{{ Changes the default fetch mode for subsequent fetches (params have different meaning for different fetch modes) */
 
-int pdo_stmt_setup_fetch_mode(pdo_stmt_t *stmt, zend_long mode, zval *args, uint32_t num_args)
+bool pdo_stmt_setup_fetch_mode(pdo_stmt_t *stmt, zend_long mode, uint32_t mode_arg_num,
+	zval *args, uint32_t variadic_num_args)
 {
 	int flags = 0;
-	zend_class_entry *cep;
-	int retval;
-
-	do_fetch_opt_finish(stmt, 1);
+	uint32_t arg1_arg_num = mode_arg_num + 1;
+	uint32_t constructor_arg_num = mode_arg_num + 2;
+	uint32_t total_num_args = mode_arg_num + variadic_num_args;
+	zval arg1, constructor_args;
+
+	/* Assign variadic params to dedicated ones */
+	if (variadic_num_args >= 1) {
+		arg1 = args[0];
```

Comment:
Why write `arg1` instead of `args[0]`?

Reply:
Was trying to get it to work ASAP, but yeah makes more sense to use it directly